### PR TITLE
Remove Docker Quickstart and Teleport-Lab Image

### PR DESCRIPTION
**Pull request by benarent**
We no longer use Docker Quickstart in Docs or goteleport.com/labs.

I've not removed `teleport-lab` but after Drone migration we might want to consider sunsetting this Docker file and instead focusing `Docker` for people running our images against Teleport Cloud

### CODE REVIEW

File: docs/postrelease.md [hunk 1]
```diff
@@ -21,7 +21,6 @@ is published, since the PR will include an update to the plugins version as well
 - [ ] Update `branchMajorVersion` const in Dronegen `/dronegen/container_images.go`, then run `make dronegen`
 - [ ] Update `CURRENT_VERSION_ROOT`, `PREVIOUS_VERSION_ONE_ROOT`, and `PREVIOUS_VERSION_TWO_ROOT` variables in `.drone.yml`, then run `make dronegen`
   - Example: https://github.com/gravitational/teleport/pull/4602
-- [ ] Create PR to update default Teleport image referenced in docker/teleport-quickstart.yml
   - Example: https://github.com/gravitational/teleport/pull/4655
```
Comment (reedloden) [inline docs/postrelease.md:24]:
```suggestion
```

**Review by reedloden — APPROVED** [review]
One nit

**Comment by stevenGravy** [comment]
Once we have confirmation there are no depends I would just remove the whole docker directory. Things like the sshd dockerfile can't work with the distroless version.

**Review by zmb3 — APPROVED** [review]
LGTM assuming Reed's comment is addressed.

**Comment by benarent** [comment]
I've added a few more commit to remove teleport-lab drone image. There is some argument for removing the folder completely and updating it with a readme for the distroless docker images. I'll plan to do that in another PR.

**Review by thedevelopnik — APPROVED** [review]

**Review by flyinghermit — APPROVED** [review]

**Pull request merged**